### ISSUE TITLE
Two identical Flexviews setups, but I'm seeing different `dml_type` values?

### ISSUE THREAD

**Issue by korlaxxalrok** [greenlion/swanhart-tools#168]
I have Flexviews running on two different replicas that should are tracking changes from the same datasoure. My new setup mostly seems to work, but I'm seeing different `dml_type` values.

Where I'd expect to see `2`, I'm seeing instead a `dml_type` ranging from `8` to `4096`.

Anyone have a thought on what might be causing this?

**Comment by greenlion** [comment]
I don't see how that is possible.  What version of MySQL/MariaDB are you
using?  If you copy the Flexviews installation from the old host to the new
do you have the problem?

**Comment by greenlion** [comment]
Also, dml_type should only be two if some special configuration options are set.  Can you please post your config.  You can also turn on SQL debugging by updating include/flexcdc.php and changing $debug = false to $debug = true in the function that executes SQL statements (just search for $debug and update the default of the variable...)

Give me the output of the mysqlbinlog command (should be printed by FlexCDC) along with the SQL output.

Thanks!

**Comment by korlaxxalrok** [comment]
Let me start fresh and clone the working installation and set things up again. I'll grab debug info after that.

Re the weird dml_type values, I think this is related to what I had mentioned in my other issue. I was watching the query log and it looks like the dml_type would start at `2`, then double for each insert statement. Eventually it would become large enough that it would start being represented in exponent notation.

We do have `mark_updates=true`, which is:
```
;FlexCDC normally marks INSERTS as a dml_type=1 and DELETEDS as dml_type=-1.
;Some ETLs need to know that a insert or delete is from an update statement.
;set mark_updates=true if you want the dml_type for INSERTS to be 2 and for DELETES -2.
;SETTING THIS OPTION *IS NOT COMPATIBLE WITH FLEXVIEWS MATERIALIZED VIEWS*!!!!!!!!!
;ONLY SET THIS OPTION IF USING FLEXCDC STANDALONE AND YOU KNOW WHAT YOU ARE DOING
```
I don't actually have the history on why this setting was chosen, but it appears to be working on our existing setup.

Let me clean things up and start fresh, though.

**Comment by greenlion** [comment]
Uh, are you sure that is the comment in the file.  Inserts should be -1, delete +1 and updates 2 if mark_updates is true.

**Comment by greenlion** [comment]
Sorry for the confusion.  So, I can't see how the dml_type would be incrementing.  It is hard coded to -2 for the BEFORE image, +2 for the AFTER image for updates, which is what I meant.

Maybe an ALTER somehow messed up the order of rows in the mvlog table?  In addition to the above output, give me the SHOW CREATE TABLE for the log table.

**Comment by korlaxxalrok** [comment]
Alrighty, I spent a bit of time more faithfully producing our current and working configuration.

**Things are looking good now!**

I did have to get set up an on older instance of Ubuntu that had an older version of PHP available from apt. Once I set up the dependencies and copied the build from our current and working system I'm getting the `-2/2` values as expected.

I'm not sure if there was something strange in the config on the new system I was setting up, but I had tried both PHP 5.6 and 7.2 against the latest Flexviews clone from the repo, as well as PHP 7.2 against the build from our current working system, and in both cases I was getting the strangely doubled `dml_typ` values. I actually had an issue running the Flexviews build from our current working system with PHP 5.6 on the new system (because this was a very recent Ubuntu I had to fiddle around to get PHP 5.6 installed and I suspect it was not totally "clean" was therefor getting some blank stares when I tried to run it), so I wasn't able to test and troubleshoot as fully as I wanted to.

But after getting an older Ubuntu + solid PHP 5.5.9 + current working Flexviews build set up things are looking good.

I think I'll leave it there for now because it is running and now I can proceed with the rest of our migration out of Joyent :)

I appreciate you dropping in to offer support! Can I buy you a coffee? ☕️ ☕️ ☕️
